### PR TITLE
Mark code in `typescript.md`

### PR DESCRIPTION
**Pull request by hippotastic**
#### What kind of changes does this PR include?

- New or updated content

#### Description

- Uses our new code snippet inline marking feature to highlight important parts of the code blocks.

### CODE REVIEW

File: src/pages/en/guides/typescript.md [hunk 1]
```diff
@@ -87,7 +87,7 @@ import Layout from '@layouts/Layout.astro';
 
 Astro supports typing your component props via TypeScript. To enable, export a TypeScript `Props` interface from your Astro component. The [Astro VSCode Extension](/en/editor-setup/) will automatically look for the `Props` export and give you proper TS support when you use that component inside another template.
 
-```astro title="src/components/HelloProps.astro"
+```astro title="src/components/HelloProps.astro" {2-5} "as Props"
```
Comment (Pukimaa) [inline src/pages/en/guides/typescript.md:90]: What about green highlighting here?

File: src/pages/en/guides/typescript.md [hunk 2]
```diff
@@ -102,7 +102,7 @@ const { greeting = 'Hello', name } = Astro.props as Props;
 
 Astro provides JSX type definitions to check that your markup is using valid HTML attributes. You can use these types to help build component props. For example, if you were building a `<Link>` component, you could do the following to mirror the default HTML attributes in your component’s prop types.
 
-```astro title="src/components/Link.astro"
+```astro title="src/components/Link.astro" {2} "as Props"
```
Comment (Pukimaa) [inline src/pages/en/guides/typescript.md:105]: Same applies here

**Review by Pukimaa — CHANGES_REQUESTED** [review]
Great Work! Just one little change I would personally do.

File: src/pages/en/guides/typescript.md (hunk 2 above)
Comment (hippotastic) [reply src/pages/en/guides/typescript.md:105]: I agree to both of your suggestions! Will change that :)

**Pull request merged**